src/services/nativeAuth/helpers/getLatestBlockHash.ts

```diff
@@ -23,8 +23,17 @@ const requestPromise: {
 const getLatestBlockHashFromServer = retryMultipleTimes(
   async (
     apiUrl: string,
-    blockHashShard?: number
+    blockHashShard?: number,
+    getBlockHash?: () => Promise<string>
   ): Promise<LatestBlockHashType | null> => {
+    // get current block hash
+    if (getBlockHash) {
+      const timestamp = Math.floor(Date.now() / 1000);
+      const hash = await getBlockHash();
+
+      return { hash, timestamp };
+    }
+
     //get the penultimate block hash (3 shards + the meta chain) to make sure that the block is seen by auth server
     const { data } = await axios.get<Array<LatestBlockHashType>>(
```

Comment:
if we pass` nativeAuthClient.getCurrentBlockHash()` anyway, is it still necessary to keep this code?
in this case we can make it mandatory